Grbl_Esp32/src/Motors/Motor.h

```diff
@@ -36,20 +36,21 @@
 namespace Motors {
     class Motor {
     public:
-        Motor();
+        Motor(motor_class_id_t type, uint8_t axis_index);
 
-        virtual void init();  // not in constructor because this also gets called when $$ settings change
-        virtual void config_message();
-        virtual void debug_message();
-        virtual void read_settings();
-        virtual void set_homing_mode(uint8_t homing_mask, bool isHoming);
-        virtual void set_disable(bool disable);
-        virtual void set_direction_pins(uint8_t onMask);
-        virtual void step(uint8_t step_mask, uint8_t dir_mask);  // only used on Unipolar right now
-        virtual bool test();
-        virtual void set_axis_name();
-        virtual void update();
-        virtual bool can_home();
+        virtual void  init();  // not in constructor because this also gets called when $$ settings change
+        virtual void  config_message();
+        virtual void  debug_message();
+        virtual void  read_settings();
+        virtual void  set_homing_mode(uint8_t homing_mask, bool isHoming);
+        virtual void  set_disable(bool disable);
+        virtual void  set_direction(bool);
+        virtual void  step();
+        virtual void  unstep();
```

Comment:
What would have helped me while reading this code is a brief explanation about what 'unstep' / 'step' are (comment) and how they are supposed to be used from the outside code. What I'm guessing is that you need a 'step'/'unstep' pair to do a single step. But it might also mean that you do a lot of steps, and then 'unstep' when you're done. Or that 'unstep' is some sort of reset.

Reply:
(After reading the rest of the code) I'm also still uncertain how it relates to 'set_direction'. It appears as if they are tightly working together, but I'm not certian how. More specific: is set_direction only used while motors are unstepped?

It might even be good to put some assertions in the (null) motor to check this required behavior. That allows us to catch bugs before they are committed.

Reply:
It's tricky because of the rather-strange way that Stepper.cpp is implemented.  It needs more attention and checking.